src/components/calcite-date-month-header/calcite-date-month-header.tsx

```diff
@@ -0,0 +1,249 @@
+  h,
+  Watch
+} from "@stencil/core";
+import { ENTER, SPACE } from "../../utils/keys";
+
+@Component({
+  tag: "calcite-date-month-header",
+  styleUrl: "calcite-date-month-header.scss",
+  shadow: true
+})
+export class CalciteDateMonth {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * Month number starting 0 as January for which the calendar is shown.
+   */
+  @Prop() month: number = 0;
+  /**
+   * Year for which the calendar is shown.
+   */
+  @Prop() year: number = 0;
+  /**
+   * Already selected date.
+   */
+  @Prop() selectedDate: Date;
+  /**
+   * Minimum date of the calendar below which is disabled.
+   */
+  @Prop() min: Date;
+  /**
+   * Maximum date of the calendar above which is disabled.
+   */
+  @Prop() max: Date;
+  /**
+   * pass the locale in which user wants to show the date.
+   */
+  @Prop() locale: string = "en-US";
+  /**
+   * Localized string for previous month.
+   */
+  @Prop() prevMonthLabel: string = "";
+  /**
+   * Localized string for next month.
+   */
+  @Prop() nextMonthLabel: string = "";
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+  /**
+   *  Event triggered when user change month.
+   */
+  @Event() calciteMonthChange: EventEmitter;
+  /**
+   *  Event triggered when user change year.
+   */
+  @Event() calciteYearChange: EventEmitter;
+
+  @Watch("month") monthChange() {
+    this.calciteMonthChange.emit();
+  }
+
+  @Watch("year") yearChange() {
+    this.calciteYearChange.emit();
+  }
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  componentWillUpdate(): void {}
+
+  render() {
+    let localizedMonth = this.getLocalizedMonths()[this.month];
+
+    return (
+      <Host>
+        <div class="month-year" aria-hidden="true">
+          <span
+            role="button"
+            aria-label={this.prevMonthLabel}
+            tabindex={0}
+            onClick={() => this.selectPrevMonth()}
+            onKeyDown={event => this.selectPrevMonthOnEnter(event)}
+          >
+            <svg
+              xmlns="http://www.w3.org/2000/svg"
+              class="left-icon"
+              viewBox="0 0 16 16"
+              height="16"
+              width="16"
+            >
+              <path d="M11.783 14H9.017l-6-6 6-6h2.766l-6 6z" />
+            </svg>
+          </span>
+          <div>
+            <span class="month" role="heading">
```

Comment:
why not use a heading?

Reply:
The main reason I used span is it is inline and the dom here is needed inline dom else i have to style those again. :(